libraries/AC_AttitudeControl/AC_PosControl.h

```diff
@@ -45,204 +45,303 @@ class AC_PosControl
 
-    // relax_U_controller - initialise the position controller to the current position and velocity with decaying acceleration.
-    ///     This function decays the output acceleration by 95% every half second to achieve a smooth transition to zero requested acceleration.
+    // Smoothly decays U-axis acceleration to zero over time while maintaining current vertical velocity.
+    // Reduces requested acceleration by ~95% every 0.5 seconds to avoid abrupt transitions.
+    // `throttle_setting` is used to determine whether to preserve positive acceleration in low-thrust cases.
     void relax_U_controller(float throttle_setting);
 
-    // init_U_controller - initialise the position controller to the current position, velocity, acceleration and attitude.
-    ///     This function is the default initialisation for any position control that provides position, velocity and acceleration.
-    ///     This function is private and contains all the shared z axis initialisation functions
+    // Fully initializes the U-axis controller with current position, velocity, acceleration, and attitude.
+    // Used during standard controller activation when full state is known.
+    // Private function shared by other vertical initializers.
     void init_U_controller();
 
-    /// input_accel_U_cm - calculate a jerk limited path from the current position, velocity and acceleration to an input acceleration.
-    ///     The function takes the current position, velocity, and acceleration and calculates the required jerk limited adjustment to the acceleration for the next time dt.
-    ///     The kinematic path is constrained by the maximum acceleration and jerk set using the function set_max_speed_accel_z.
+    // Sets the desired vertical acceleration in cm/s² using jerk-limited shaping.
+    // See input_accel_U_m() for full details.
     virtual void input_accel_U_cm(float accel_u_cmss);
 
-    /// input_vel_accel_U_cm - calculate a jerk limited path from the current position, velocity and acceleration to an input velocity and acceleration.
-    ///     The function takes the current position, velocity, and acceleration and calculates the required jerk limited adjustment to the acceleration for the next time dt.
-    ///     The kinematic path is constrained by the maximum acceleration and jerk set using the function set_max_speed_accel_z.
-    ///     The function modifies vel_u_cms to follow the jerk-limited trajectory defined by accel_u_cmss.
-    ///     The parameter limit_output specifies if the velocity and acceleration limits are applied to the sum of commanded and correction values or just correction.
+    // Sets the desired vertical acceleration in m/s² using jerk-limited shaping.
+    // Smoothly transitions to the target acceleration from current kinematic state.
+    // Constraints: max acceleration and jerk set via set_max_speed_accel_U_m().
+    virtual void input_accel_U_m(float accel_u_mss);
```

Comment:
not a blocker because it's an existing issue but I think we can remove the "virtual" from before input_accel_U_cm and input_accel_U_m